util/compiler.h

```diff
@@ -5,11 +5,13 @@
 /* Use to tag a variable that causes compiler warnings. Use as:
     int uninitialized_var(sz)
 
-   This is only enabled for old compilers. gcc 6.x and beyond have excellent
-   static flow analysis. If code solicits a warning from 6.x it is almost
-   certainly too complex for a human to understand.
+   This is only enabled for old compilers and non-x86 architectures. gcc 6.x
+   and beyond have excellent static flow analysis for x86 platforms. If code
+   solicits a warning from 6.x it is almost certainly too complex for a human to
+   understand.
 */
-#if __GNUC__ >= 6 || defined(__clang__)
+#if (__GNUC__ >= 6 && !defined(__aarch64__)) || \
+    defined(__clang__)
 #define uninitialized_var(x) x
```

Comment:
Please see below:
```
../providers/mthca/qp.c: In function ‘mthca_arbel_post_send’:
../providers/mthca/qp.c:448:29: error: ‘f0’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  uint32_t uninitialized_var(f0);
                             ^~
../providers/mthca/qp.c:449:29: error: ‘op0’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  uint32_t uninitialized_var(op0);
                             ^~~
cc1: all warnings being treated as errors
[44/92] Building C object providers/mlx5/CMakeFiles/mlx5.dir/verbs.c.o
```

I can check with compiler people but my guess is that the static analysis flow is not the same.